lib/run-status.js

```diff
@@ -18,10 +18,15 @@ function sum(arr, key) {
 }
 
 function normalizeError(err) {
-	if (!isObj(err)) {
+	if (isObj(err) && 'message' in err && typeof err.message === 'string') {
+		['name', 'stack'].forEach(field => {
+			if (field in err && typeof err[field] !== 'string') {
+				delete err[field];
```

Comment:
I think this really does need to go here: https://github.com/avajs/ava/blob/98dded553c8c213739cea9743ab4662f40d3125a/lib/serialize-error.js#L59

We don't want to modify `err` since it comes from user code. There may be side-effects. But we can't easily copy error objects since `name` and `message` aren't always enumerable. After the call to `cleanYamlObject` in `serialize-error.js` we have a copy we can safely modify.

Also, in this case, I'd prefer a little bit of duplication over your `forEach` trick. IMHO this is a lot easier to follow:

```js
if ('name' in err && typeof err.name !== 'string') {
  delete err.name
}
if ('stack' in err && typeof err.stack !== 'string') {
  delete err.stack
}
```

Even better, `delete` only deletes own-properties, so the `'name' in err` guard is unnecessary.